client-app/src/common/models/search/terms.ts

```diff
@@ -0,0 +1,38 @@
+import { TermsItems } from './terms-items';
+
+export class Terms {
+    items: TermsItems = {};
+
+    toggle(key: string, value: string) {
+      // if key & value exists
+      if (this.items[key]?.includes(value)){
+        // remove value
+        this.items[key].splice(this.items[key].indexOf(value), 1);
```

Comment:
I prefer a more clean approach, eg:
![image](https://user-images.githubusercontent.com/12604378/78007539-a1689480-7368-11ea-9695-3115fd20bbb9.png)

[example.txt](https://github.com/VirtoCommerce/vc-odt-mpa-theme/files/4407812/example.txt)

1. in this way your code more flexible for reusing
2. no need extra comments
3. you manipulate methods like verbs
4. your code decompose to small pieces
etc

Reply:
Refactored